Akkadu/rsi-api-widgets
# rsi-vanilla-js-package-and-automatize-versioning

Please describe the purpose of the pull request.

* The purpose was to make the vanilla javascript Widget
* Build a vanilla widget [notion](https://www.notion.so/akkadu/RSI-API-Interpretation-Player-VanillaJs-0d77311ccd814474afb29ea136e1eb6d)
* Automatize versioning [notion](https://www.notion.so/akkadu/RSI-API-Interpretation-Player-VanillaJs-Automatic-versionning-11c7a32cb2ae415181d090401cf2c6ac)

## Motivation and Context
* I made the vanilla javascript widget : `rsi-api-vanilla`
* I made a demo for the client : `rsi-api-vanilla-demo`
* Updated the [doc](https://deploy-preview-12--rsi-akkadu-documentation.netlify.app/getting-started/vanilla-js.html)
* Built a script to automatically deploy on AWS `akkadu-assets` the new version of vanilla.
  * This script will run when doing `lerna publish`
  * That script also update the documentation : vanilla/versions : https://deploy-preview-12--rsi-akkadu-documentation.netlify.app/vanilla-js/versions.html. 


## How To Test
* In the root folder : `yarn`, then `yarn build`
* Go to the folder : rsi-api-vanilla-demo` and start the index.html in a live server.
  * On vscode you should have an extenxion to start live servers. 
  * Make sure the url is `localhost` (otherwise you will get a core issue, we will have to update that part on lambda, its currently not accepting `http://127.0.0.1/`)
  * You should be able to use the widget and switch between languages.


## Additional
* for that package rsi-api-vanilla: i'm using webpack instead of rollup, i was not able with rollup to use babel on the node_dependencies of the `interpretation-player`.



## Screenshots (if appropriate)

## Checklist



加油！
